br/pkg/lightning/restore/get_pre_info.go

```diff
@@ -590,11 +598,21 @@ func (p *PreRestoreInfoGetterImpl) sampleDataFromTable(
 		return resultIndexRatio, isRowOrdered, nil
 	}
 	sampleFile := tableMeta.DataFiles[0].FileMeta
-	var reader storage.ReadSeekCloser
-	var err error
-	if sampleFile.Type == mydump.SourceTypeParquet {
+	var (
+		reader storage.ReadSeekCloser
+		err    error
+	)
+	switch {
+	case sampleFile.Type == mydump.SourceTypeParquet:
 		reader, err = mydump.OpenParquetReader(ctx, p.srcStorage, sampleFile.Path, sampleFile.FileSize)
-	} else {
+	case sampleFile.Compression != mydump.CompressionNone:
+		var compressType storage.CompressType
+		compressType, err = mydump.ToStorageCompressType(sampleFile.Compression)
+		if err != nil {
+			break
+		}
+		reader, err = storage.WithCompression(p.srcStorage, compressType).Open(ctx, sampleFile.Path)
```

Comment:
Duplicate with `reader.go` too. Maybe we should encapsulate this with a function.

Reply:
addressed in https://github.com/pingcap/tidb/pull/39153/commits/4c7eaf9ef4e4b6df835fcd939d1325f86b2801b6